src/CoreBundle/Bridge/Chameleon/Module/NavigationTreeSingleSelect/NavigationTreeSingleSelect.php

```diff
@@ -113,27 +275,158 @@ private function getPortalTreeRootNodeId(): string
         return $portal->fieldMainNodeTree;
     }
 
-    private function createTreeDataModel(\TdbCmsTree $node, string $fieldName, $path = ''): BackendTreeNodeDataModel
+    private function createTreeDataModel(\TdbCmsTree $node, int $level, string $path = ''): BackendTreeNodeDataModel
     {
         $treeNodeDataModel = $this->backendTreeNodeFactory->createTreeNodeDataModelFromTreeRecord($node);
-        $children = $node->GetChildren(true);
+        $treeNodeDataModel->setName($this->translateNodeName($treeNodeDataModel->getName(), $node));
+        $this->setTypeAndAttributes($treeNodeDataModel, $node);
+
+        if ($level === 0) {
+            $treeNodeDataModel->setOpened(true);
+        }
 
-        $path = $this->addBreadcrumbHtmlForNode($node, $fieldName, $path);
+        //"primary-node-selection (from page)" starts with the active portal:
+        //   $level: 0 = portal, 1 = Navigation-Nodes, >1 = folder or page
 
+        //"portal-selection" and "portal-home-page-selection" and "portal-404-page-selection start with root node:
+        //   $level: 0 = rootNode (Website), 1 = portal, 2 = Navigation-Nodes, >2 folder or page
+        if (true === $this->isPortalSelectMode) {
+            if ($level !== 1) {
```

Comment:
Can you introduce (and use) static variables with appropriate names for these level signifiers? (and get rid of the comments listing them)

Reply:
ok